src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,238 @@
+                newIfNodeStatement = ifNode.Else.Statement.WithTrailingTrivia(ifTrailingTrivia);
+                newElseStatement = ifNode.Else.WithStatement(ifNode.Statement).WithTrailingTrivia(elseTrailingTrivia);
+            }
+
+            ifNode = ifNode.WithCondition(Negate(ifNode.Condition, model, cancellationToken))
+                .WithStatement(ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block ? SyntaxFactory.Block(newIfNodeStatement) : newIfNodeStatement)
+                .WithElse(newElseStatement);
+
+            if (isMultiLine)
+            {
+                ifNode = ifNode.WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            // get new root
+            return model.SyntaxTree.GetRoot().ReplaceNode(ifStatement, ifNode);
+        }
+
+        private bool TryNegateBinaryComparisonExpression(
+            ExpressionSyntax expression,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken,
+            out ExpressionSyntax result)
+        {
+            if (s_binaryMap.TryGetValue(expression.Kind(), out var negatedExpressionInfo))
+            {
+                var binaryExpression = (BinaryExpressionSyntax)expression;
+                var (negatedExpressionType, negatedOperatorType) = negatedExpressionInfo;
+
+                // Certain expressions can never be negative, such as length or unsigned numeric types.
+                // If these expressions are compared with zero using <, >, or =, we construct the negated
+                // binary expression to reflect that it will never be negative.
+                // For example, the expression Array.Length > 0, becomes Array.Length == 0 when negated.
+                var operation = semanticModel.GetOperation(binaryExpression);
+
+                if (IsSpecialCaseBinaryExpression(operation as IBinaryOperation, cancellationToken))
+                {
+                    negatedOperatorType = binaryExpression.OperatorToken.Kind() == SyntaxKind.EqualsEqualsToken
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanToken : SyntaxKind.LessThanToken
+                        : SyntaxKind.EqualsEqualsToken;
+                    negatedExpressionType = binaryExpression.Kind() == SyntaxKind.EqualsExpression
+                        ? binaryExpression.Right is LiteralExpressionSyntax ? SyntaxKind.GreaterThanExpression : SyntaxKind.LessThanExpression
+                        : SyntaxKind.EqualsExpression;
+                }
+
+                result = SyntaxFactory.BinaryExpression(
+                    negatedExpressionType,
+                    binaryExpression.Left,
+                    SyntaxFactory.Token(
+                        binaryExpression.OperatorToken.LeadingTrivia,
+                        negatedOperatorType,
+                        binaryExpression.OperatorToken.TrailingTrivia),
```

Comment:
this pattern of making a new token, but using the old trivia is repeated many times in this file.  Please extract helper.